src/transport_udp_protocol.h

```diff
@@ -67,19 +77,33 @@ namespace qtransport {
          */
         struct KeepaliveMsg : CommonHeader {
             KeepaliveMsg() { type = ProtocolType::KEEPALIVE; }
+
+            uint16_t ticks_ms { 0 };            /// Senders Tick millisecond value from start of report period, reset to zero on new report
+
         } __attribute__((__packed__, aligned(1)));
 
         /**
          * @brief Data Message
          *
          * @details Data message. Bytes following the header is the data
          *
-         * @note Can be either type of DATA or DATA_DISCARD
+         * @note Can be either type of DATA
          */
         struct DataMsg : CommonHeader {
-            DataMsg() { type = ProtocolType::DATA; } // Can be either DATA or DATA_DISCARD type
+            DataMsg() { type = ProtocolType::DATA; }
+
+            struct {
+#if __BYTE_ORDER__ == __ORDER_BIG_ENDIAN__
+                uint8_t discard     : 1;        /// Indicates that data should be discard
+                uint8_t reserved    : 7;        /// Least significant bits reserved
+#else
+                uint8_t reserved    : 7;        /// Least significant bits reserved
+                uint8_t discard     : 1;        /// Indicates that data should be discard
+#endif
```

Comment:
Since we're using c++20, could instead use constexpr `std::endian::native == std::endian::big` instead, that would be platform agnostic.

Reply:
That works well for testing during runtime but for the struct in this form it doesn't appear to work. An alternative is:

```
include(TestBigEndian)
TEST_BIG_ENDIAN(IS_BIG_ENDIAN)
set(CXXFLAGS ${CXXFLAGS} -DIS_BIG_ENDIAN=${IS_BIG_ENDIAN})
```

In the CMakeLists.txt, but that may not be as portable as what's in there now. I'll make note of it and if we run into issues, let's try the make method.  If that becomes a problem, we can then change from memcpy style.